### PR TITLE
Split TestFunctions.py

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Please see #1199 for details.

### Types of changes:

- [x] Enhancement

### Component:

- [x] Tests

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/BondiHoyleAccretion.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+import numpy as np
+
+import PointwiseFunctions.GeneralRelativity.KerrSchildCoords as ks_coords
```
Comment (nilsvu) [inline tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/BondiHoyleAccretion.py:6]: This is how we can import functions across test modules now.

This requires `pypp::SetupLocalPythonEnvironment local_python_env{""}` in the cpp test. If no cross-imports are needed, the setup can be in a submodule, i.e. `pypp::SetupLocalPythonEnvironment local_python_env{"PointwiseFunctions/GeneralRelativity/"}`.

Comment (nilsdeppe) [reply tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/BondiHoyleAccretion.py:6]: I like this, I think it's a good approach :+1:

File: tests/Unit/PointwiseFunctions/GeneralRelativity/Test_KerrSchildCoords.cpp [hunk 2]
```diff
@@ -166,10 +165,10 @@ SPECTRE_TEST_CASE("Unit.PointwiseFunctions.GeneralRelativity.KerrSchildCoords",
 #endif
 }
 
-// [[OutputRegex, The input vector must have a vanishing theta component]]
-[[noreturn]] SPECTRE_TEST_CASE(
-    "Unit.PointwiseFunctions.GeneralRelativity.KerrSchildCoords.AlongZDv",
-    "[Unit][PointwiseFunctions]") {
+    // [[OutputRegex, The input vector must have a vanishing theta component]]
```
Comment (nilsdeppe) [inline tests/Unit/PointwiseFunctions/GeneralRelativity/Test_KerrSchildCoords.cpp]: woops, this doesn't need to be indented

File: tests/Unit/PointwiseFunctions/GeneralRelativity/Test_KerrSchildCoords.cpp [hunk 3]
```diff
@@ -188,10 +187,10 @@ SPECTRE_TEST_CASE("Unit.PointwiseFunctions.GeneralRelativity.KerrSchildCoords",
 #endif
 }
 
-// [[OutputRegex, The dimensionless spin must be in the range \(-1, 1\)]]
-[[noreturn]] SPECTRE_TEST_CASE(
-    "Unit.PointwiseFunctions.GeneralRelativity.KerrSchildCoords.SpinLower",
-    "[Unit][PointwiseFunctions]") {
+    // [[OutputRegex, The dimensionless spin must be in the range \(-1, 1\)]]
```
Comment (nilsdeppe) [inline tests/Unit/PointwiseFunctions/GeneralRelativity/Test_KerrSchildCoords.cpp]: woops, this doesn't need to be indented

**Review by nilsdeppe — COMMENTED** [review]
Overall I like this approach. I think it is a good solution :+1: thank you for doing this!

I have a couple minor formatting comments, but I'm happy with this :)

**Comment by nilsvu** [comment]
Pushed a commit updating the guidelines. I think this PR is done, and from now on we can have new code adopt these guidelines.

**Comment by nilsdeppe** [comment]
Other than every travis build failing, this looks good :) I think you have a typo somewhere in the GhQuantities tests.

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
Travis is unhappy

**Comment by nilsvu** [comment]
Looks like IWYU takes too long, can you somehow push it through Travis @nilsdeppe?

**Review by nilsdeppe — APPROVED** [review]
I ran IWYU on this locally and it passed.

**Review by carmaza — APPROVED** [review]

**Pull request merged**